config/http_config.go

```diff
@@ -172,6 +181,7 @@ func (c *HTTPClientConfig) Validate() error {
 // UnmarshalYAML implements the yaml.Unmarshaler interface
 func (c *HTTPClientConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	type plain HTTPClientConfig
+	*c = DefaultHTTPClientConfig
```

Comment:
As said, this wouldn't work if `c` were `nil`. (Which implies that `c` is not `nil` rather than that we are doing anything wrong here.)